src/ripple/overlay/Overlay.h

```diff
@@ -45,6 +47,9 @@ class Overlay
     , public beast::PropertyStream::Source
 {
 protected:
+    using socket_type = boost::asio::ip::tcp::socket;
+    using stream_type = boost::asio::ssl::stream <socket_type>;
```

Comment:
There's `beast::ssl_stream` which offers significant performance improvements for both HTTP and WebSocket purposes, by consolidating buffers (asio's SSL implementation does not support scatter/gather).

Reply:
The asio::ssl::stream class has lowest_layer() method which is used by rippled. Is there such a method in beast::ssl_stream?

Reply:
There's a free function which does the same thing
https://www.boost.org/doc/libs/1_72_0/libs/beast/doc/html/beast/ref/boost__beast__get_lowest_layer.html

Reply:
Migration to beast::ssl_stream as stream_type and beast::tcp_stream as socket_type done.